includes/config.php

```diff
@@ -0,0 +1,235 @@
+<?php
+/**
+ * Utility functions to retrieve and parse config files.
+ *
+ * @package hm-platform
+ */
+
+namespace HM\Platform\Config;
+
+use HM\Platform as Platform;
+use HM\Platform\Plugin as Plugin;
+use Exception;
+
+/**
+ * Retrieve the configuration for HM Platform.
+ *
+ * The configuration is defined by merging the defaults with the various files that allow to customise a particular
+ * installation.
+ *
+ * @throws Exception
+ * @return array Configuration data.
+ */
+function get_config() {
+	global $hm_platform;
+	static $config;
+
+	if ( ! $config ) {
+		$config = get_merged_defaults_and_customisations();
+
+		// Handle back compat for $hm_platform.
+		if ( ! empty( $hm_platform ) && is_array( $hm_platform ) ) {
+			foreach ( $hm_platform as $plugin => $enabled ) {
+				if ( isset( $config['plugins'][ $plugin ] ) ) {
+					$config['plugins'][ $plugin ]['enabled'] = $enabled;
+				}
+			}
+		}
+	}
+
+	return $config;
+}
+
+/**
+ * Return the value for a particular setting from the configuration,
+ *
+ * @param string $key Settings key to retrieve the value from.
+ *
+ * @return mixed Settings value.
+ *
+ * @throws Exception if the settings key cannot be found.
+ */
+function get_config_value( string $key ) {
+	$config = get_config();
+
+	if ( ! array_key_exists( $key, $config ) ) {
+		throw new Exception( 'Could not find the ' . $key . ' setting in the configuration.' );
```

Comment:
Do we really want all these exceptions everywhere? In production, or really anything we _really_ don't want anything in hm-platform to produce uncaught exceptions. Much of this code is only run in prod, not locally so I'm strongly against fataling via hm-platform. I'd like us to change as many functions to now throw exceptions, unless it's actually a serious error. Lets use sane return defaults (null probably here) or WP_Error where we need error objects.

Reply:
Not necessarily, though it'll be caught in `load.php`. I guess the intention was avoid potentially committing broken configs. You'd see it pretty quickly on local. Will look at updating to `null` as `WP_Error` wouldn't be defined yet.